tests/auto/sugarcrmresource/test_sugarcrmresource.cpp

```diff
@@ -0,0 +1,413 @@
+        if (name.isEmpty()) {
+            reply = mock.call(opperationName, id);
+        } else {
+            reply = mock.call(opperationName, name, id);
+        }
+        QVERIFY2(reply.isValid(), reply.error().message().toLatin1());
+        AgentManager::self()->synchronizeCollection(collection);
+
+        QSignalSpy spy(&monitor, signal);
+        QVERIFY2(kWaitForSignal(&monitor, signal, 2000), "error: TimeOut");
+
+
+        QCOMPARE(spy.count(), 1);
+        arguments = spy.at(0);
+    }
+
+    template<typename T>
+    void createSugarItem(const QString &name, const QString &id, Collection &collection, QList<ItemData> &&expected, const QString &operationName)
+    {
+        //GIVEN
+        T sugarItem;
+        sugarItem.setName(name);
+        Item item;
+        item.setPayload(sugarItem);
+        item.setMimeType(T::mimeType());
+        ItemCreateJob *job = new ItemCreateJob(item, collection);
+        //WHEN
+        AKVERIFYEXEC(job);
+        //THEN
+        Monitor monitor;
+        monitorInit(monitor, collection);
+        QSignalSpy spyAdded(&monitor, SIGNAL(itemAdded(Akonadi::Item,Akonadi::Collection)));
+        QSignalSpy spyChanged(&monitor, SIGNAL(itemChanged(Akonadi::Item,QSet<QByteArray>)));
+
+        QVERIFY(kWaitForSignal(&monitor, SIGNAL(itemAdded(Akonadi::Item,Akonadi::Collection)), 5000));
+        QVERIFY(kWaitForSignal(&monitor, SIGNAL(itemChanged(Akonadi::Item,QSet<QByteArray>)), 5000));
+
+//        for (int i = 0; i < spyAdded.count(); i++) {
+//            const QList<QVariant> arguments = spyAdded.at(i);
+//            const Item item = arguments.at(0).value<Item>();
+//            T inst = item.payload<T>();
+//            qDebug() << "TEST ////" << "added" << i << "name:" << inst.name() << "id:" << inst.id() << "remoteId:" << item.remoteId();
+//        }
+//        for (int i = 0; i < spyChanged.count(); i++) {
+//            const QList<QVariant> arguments = spyChanged.at(i);
+//            const Item item = arguments.at(0).value<Item>();
+//            T inst = item.payload<T>();
+//            qDebug() << "TEST ////" << "changed" << i << "name:" << inst.name() << "id:" << inst.id() << "remoteId:" << item.remoteId();
+//        }
+
+#if QT_VERSION >= QT_VERSION_CHECK(5, 0, 0)
+        QVERIFY(kWaitForSignal(&monitor, SIGNAL(itemChanged(Akonadi::Item,QSet<QByteArray>)), 5000));
+        QCOMPARE(spyAdded.count(), 1);
+
+        const QList<QVariant> arguments = spyAdded.at(0);
+        compareToExpectedResult<T>(arguments, name, "");
+
+        QCOMPARE(spyChanged.count(), 2);
+
+
+        const QList<QVariant> argumentsUpdate = spyChanged.at(spyChanged.count()-1);
+        compareToExpectedResult<T>(argumentsUpdate, name, id);
+#endif
+        fetchAndCompareItems<T>(collection, std::move(expected));
+
+        mNewId = spyChanged.at(0).at(0).value<Item>().id();
+        QVERIFY(mNewId != 0);
+
+        QDBusInterface mock(serviceName(), "/CRMMock", "com.kdab.SugarCRM.Mock");
```

Comment:
move interface name to a global
static const char s_dbusInterfaceName[] = "...";
to avoid the duplication.
Same for the object name.